From 9cbf452e8cc1a3ad646676bbf82e2151cb2d69ef Mon Sep 17 00:00:00 2001
From: Takayuki 'January June' Suwa <jjsuwa_sys3175@yahoo.co.jp>
Date: Wed, 24 Jul 2024 06:07:06 +0900
Subject: [PATCH] xtensa: Add missing speed cost for TYPE_FARITH in
 TARGET_INSN_COST

According to the implemented pipeline model, this cost can be assumed to be
1 clock cycle.

gcc/ChangeLog:

	* config/xtensa/xtensa.c (xtensa_insn_cost):
	Add a case statement for TYPE_FARITH.
---
 gcc/config/xtensa/xtensa.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/config/xtensa/xtensa.c b/gcc/config/xtensa/xtensa.c
index 0dd65500f..4d5e07b92 100644
--- a/gcc/config/xtensa/xtensa.c
+++ b/gcc/config/xtensa/xtensa.c
@@ -4759,6 +4759,7 @@ xtensa_insn_cost (rtx_insn *insn, bool speed)
 	    case TYPE_ARITH:
 	    case TYPE_MULTI:
 	    case TYPE_NOP:
+	    case TYPE_FARITH:
 	    case TYPE_FSTORE:
 	      return COSTS_N_INSNS (n);
 
-- 
2.39.5

